fabric-rendering-v1/src/main/java/net/fabricmc/fabric/mixin/client/rendering/MixinWorldRenderer.java

```diff
@@ -170,4 +171,40 @@ private void afterRender(CallbackInfo ci) {
 	private void onReload(CallbackInfo ci) {
 		InvalidateRenderStateCallback.EVENT.invoker().onInvalidate();
 	}
+
+	@Inject(at = @At("HEAD"), method = "renderWeather", cancellable = true)
+	private void renderWeather(LightmapTextureManager manager, float tickDelta, double x, double y, double z, CallbackInfo info) {
+		EnvironmentRenderers.WeatherRenderer renderer = null;
+
+		if (this.client.world != null) renderer = EnvironmentRenderers.getWeatherRenderer(world.getRegistryKey());
+
+		if (renderer != null) {
+			renderer.render(this.client, manager, tickDelta, x, y, z);
+			info.cancel();
+		}
+	}
+
+	@Inject(at = @At("HEAD"), method = "renderClouds(Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/util/math/Matrix4f;FDDD)V", cancellable = true)
+	private void renderCloud(MatrixStack matrices, Matrix4f matrix4f, float tickDelta, double cameraX, double cameraY, double cameraZ, CallbackInfo info) {
+		EnvironmentRenderers.CloudRenderer renderer = null;
+
+		if (this.client.world != null) renderer = EnvironmentRenderers.getCloudRenderer(world.getRegistryKey());
+
+		if (renderer != null) {
+			renderer.render(this.client, matrices, matrix4f, tickDelta, cameraX, cameraY, cameraZ);
```

Comment:
Would it be useful if the renderer returned a boolean to decide if the vanilla logic should still be used? Would make it useful to render stuff alongside the vanilla stuff?

Reply:
Hmmmmmmmm. I just reread your stuff. That could be interesting. I'm however gonna add you don't have to get an API 100%perfect the first time. Iteration is a thing after all.